cmd/moduleversion/main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	repotools "github.com/awslabs/aws-go-multi-module-repository-tools"
+	"github.com/awslabs/aws-go-multi-module-repository-tools/changelog"
+	"github.com/awslabs/aws-go-multi-module-repository-tools/git"
+	"github.com/awslabs/aws-go-multi-module-repository-tools/gomod"
+	"github.com/awslabs/aws-go-multi-module-repository-tools/release"
+)
+
+var (
+	getUnreleasedVersion bool
+)
+
+func init() {
+	flag.BoolVar(&getUnreleasedVersion, "unreleased", false,
+		"Returns the version the projected version the module will be at after the next release")
+
+	flag.Usage = func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "Usage of %s [-unreleased] module\n", filepath.Base(os.Args[0]))
+		fmt.Fprintf(flag.CommandLine.Output(), "  module\n\tThe relative path of the module to get the version of.\n")
+		flag.PrintDefaults()
+	}
+}
+
+func main() {
+	flag.Parse()
+
+	if len(flag.Args()) != 1 {
+		flag.Usage()
+		log.Fatalf("no module specified")
+	}
+	moduleToCheck := flag.Args()[0]
+
+	repoRoot, err := repotools.GetRepoRoot()
+	if err != nil {
+		log.Fatalf("failed to get repository root: %v", err)
+	}
+
+	config, err := repotools.LoadConfig(repoRoot)
+	if err != nil {
+		log.Fatalf("failed to load repotools config: %v", err)
+	}
+
+	discoverer := gomod.NewDiscoverer(repoRoot)
+
+	if err := discoverer.Discover(); err != nil {
+		log.Fatalf("failed to discover repository modules: %v", err)
+	}
+
+	tags, err := git.Tags(repoRoot)
+	if err != nil {
+		log.Fatalf("failed to get git tags: %v", err)
+	}
+
+	taggedModules := git.ParseModuleTags(tags)
+
+	annotations, err := changelog.GetAnnotations(repoRoot)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	checkedModules, err := release.Calculate(discoverer, taggedModules, config, annotations)
+	if err != nil {
+		log.Fatalf("failed to check repo modules, %v", err)
+	}
+
+	if getUnreleasedVersion {
+		id := release.NextReleaseID(tags)
+		manifest, err := release.BuildReleaseManifest(discoverer.Modules(), id, checkedModules, false)
+		if err != nil {
+			log.Fatalf("failed to build release manifest, %v", err)
+		}
+
+		if m, ok := manifest.Modules[moduleToCheck]; ok {
+			moduleTag, err := git.ToModuleTag(moduleToCheck, m.To)
+			if err != nil {
+				log.Fatalf("failed to get module %v tag, %v", moduleToCheck, err)
+			}
+			fmt.Println(moduleTag)
+			return
+		}
+	}
+
+	checkedModule, ok := release.FindModuleViaRelativeRepoPath(checkedModules, moduleToCheck)
+	if !ok {
+		log.Fatalf("failed to find version for module, %v", moduleToCheck)
+	}
+
+	moduleVersion := checkedModule.Latest
+	if checkedModule.Latest == "" {
+		moduleVersion = "v0.0.0-00010101000000-000000000000"
```

Comment:
This should technically be vN where N >= 2. Not sure if the tooling complains otherwise though.

Reply:
Cool wasn't sure the best value to return in this case. This branch would happen if the module hasn't been tagged at all yet.  Alternative to a canned value was returning an error.